### ISSUE TITLE
Upstart Service Provider doesn't handle certain status output

### ISSUE THREAD

**Issue by tarrall** [chef/chef#2819]
Back in May I mentioned in CHEF-5265 that the "status" output isn't well-parsed for some Upstart jobs; now that I've been 

Some Upstart services, including "network-interface", take parameters.  Some of them also have hyphens in their names.  In both cases, the UPSTART_STATE_FORMAT regex in Chef::Provider::Service::Upstart fails to match the output of "status <service>".

The [status(8) man page](http://manpages.ubuntu.com/manpages/trusty/man8/initctl.8.html) and [Ubuntu Upstart docs](http://upstart.ubuntu.com/cookbook/#initctl-status) give the following possible status output:

````
job start/pre-start, process 902

job start/post-start, process 1234
      post-start process 1357

job start/post-start, (post-start) process 1357

job (tty1) start/post-start, process 1234
      post-start process 1357
````

In addition, I see this output from "status network-interface INTERFACE=eth0":
````
network-interface (eth0) start/running
````

The current regex is :
    UPSTART_STATE_FORMAT = /\w+ \(?(\w+)\)?[\/ ](\w+)/

The "\w" doesn't match hyphenated job names, and the regex doesn't work when there's an instance (like "eth0" in the status output.  I think that this:

    UPSTART_STATE_FORMAT = /\S+ (?:\(\S+\) )?([\w-]+)\/([\w-]+)/

would work.  Anyone have counterexamples where this won't work, or shall I go ahead and submit a PR?  (CLA already signed.)

**Comment by lamont-granquist** [comment]
PR would be useful to get it fixed.  In general this bites us all the time and regexps should err on the side of being permissive.  I think your regexp looks fine, but I've also got a headcold so I might be missing something.

**Comment by thommay** [comment]
@tarrall hey, we would very much like the PR if you'd still like to submit one.

**Comment by tas50** [comment]
This was fixed by #4160
